### ISSUE TITLE
Pokemon Crystal GBC

### ISSUE THREAD

**Issue by dubsalicious** [mgba-emu/mgba#348]
This is a slight guess for replicating, but this appears to be what occurred.
- Capture 6 Pokemon (including starter pokemon)
- Capture a 7th Pokemon, when the screen pops up for giving it a nickname an image of the pokemon pops up for a moment and the game reboots.
- Pops up with a screen saying "This Game Pak is designed only for use on the Game Boy Color"
- Can't do anything, no inputs work, have to reboot the emulator

As mentioned elsewhere, this might be more a feature request due to the lack of official support for GBC, but was told to report it.

I have all the files to replicate at the moment, I'll aim to keep them in case I can help at all.

**Comment by endrift** [comment]
I'll take a look at this soon. If you could upload the save game (.sav file), that would save me some time.

**Comment by charsent** [comment]
No .sav I'm afraid (the copy I have seems to have been blanked somehow, its 0KB). However I've got 4 savestates, all of which are around the issue. One of them is just throwing the poke ball on the pokemon, where the crash occurs.

Also, I tried with the first release and then the nightly called "mGBA-nightly-2016-08-15-vita-7212854cdd33c1fc273cd9d451a538f4e7724e7a" in case that helps.

[savestates.zip](https://github.com/mgba-emu/mgba/files/421542/savestates.zip)

**Comment by endrift** [comment]
Ok, I've managed to reproduce.

**Comment by endrift** [comment]
The problem looks like mGBA is somehow corrupting the savedata, and it's crashing while trying to parse the savedata. BGB does the same thing with the savedata.

**Comment by msmigiel** [comment]
I have a similar issue with Pokemon Crystal and capturing more than 6 Pokemon which may be related to this.
1.) Capture 6 Pokemon to fill your party.
2.) Capturing a 7th Pokemon means it will be stored on the PC. Sometimes when this happens, the screen is garbled but says it has been stored. However, even if there are no crashes and the screen is not garbled, the 7th Pokemon never gets into the PC. I unfortunately didn't notice this until I captured about 12 total Pokemon and realized all but the first 6 were missing.
3.) Similarly, with only 6 Pokemon, storing one in the PC, leaving the PokeCenter, and save stating, upon reloading the save state and going back into the PokeCenter, the stored Pokemon is almost always gone.

**Comment by endrift** [comment]
That's almost certainly an issue with the savestate and the savegame getting out of sync. Is this still an issue with the latest nightly, or were you using 0.5.0?

**Comment by msmigiel** [comment]
I've tried nightly build "2016-10-06-win32-b5e706981fc1798163f9612441eae4df893b8192" but am noticing two major problems that prevent me from simulating my original issue.

1.) Whenever I save the game via the Save menu, mGBA crashes.
2.) Whenever I am in a Pokemon battle and whenever the first hit lands (excluding misses), mGBA crashes.

I've attached a save state
[Pokemon - Crystal Version (UE) (V1.1) [C][!].ss1.zip](https://github.com/mgba-emu/mgba/files/517642/Pokemon.-.Crystal.Version.UE.V1.1.C.ss1.zip)
 here in case it's a problem with the save state.
